conditions/operator_conditions.go

```diff
@@ -32,9 +32,15 @@ var (
 	ErrNoOperatorCondition = fmt.Errorf("operator Condition CRD is nil")
 )
 
-// TODO: verify from OLM if this will be the name of the environment variable
-// which is set for the Condition resource owned by the operator.
-const operatorCondEnvVar = "OPERATOR_CONDITION_NAME"
+const (
+	// operatorCondEnvVar is the env variable which
+	// contains the name of the Condition CR associated to the operator,
+	// set by OLM.
+	operatorCondEnvVar = "OPERATOR_CONDITION_NAME"
+	// operatorNsEnvVar is the env variable which can be used to set the namespace
+	// of the operator if running locally.
+	operatorNsEnvVar = "OPERATOR_NAMESPACE_LOCAL"
```

Comment:
If an operator is running locally, then it seems pretty unlikely that OLM will know anything about it, and if that's true, there won't be an OperatorCondition object anyway.

IMO, we should:
1. Have `GetNamespacedName()` return a checkable errors (i.e. errors that can be checked with `errors.Is()` or `errors.As()`) if it cannot find the namespace file or the operator condition environment variable.
2. Put good GoDoc comments on `GetNamespacedName()` that explains when it works (generally only in a cluster being managed by OLM), what errors callers should expect when it fails, and maybe an example of how to handle those errors.

If I we're writing an operator using this, I think I would call this function all the time, and then gracefully handle the errors by logging a message and/or setting some configuration value like `isOLMDeployed := false`, which could be used to disable the code that manages the `OperatorCondition`.

How about something like:

```go
type ErrGetNamespacedName struct {
	Error error
}

func (e ErrGetNamespacedName) Error() string {
	return fmt.Sprintf("could not determine operator condition namespaced name: %v", e.Error)
}

func GetNamespacedName() {
	conditionName := os.Getenv(operatorCondEnvVar)
	if conditionName == "" {
		return nil, &ErrGetNamespacedName{
			Error: fmt.Errorf("environment variable %s not set", operatorCondEnvVar),
		}
	}
	operatorNs, err := getOperatorNamespace()
	if err != nil {
		return nil, &ErrGetNamespacedName{
			Error: fmt.Errorf("could not determine operator namespace: %v", err),
		}
	}
	return &types.NamespacedName{Name: conditionName, Namespace: operatorNs}, nil
}
```

Reply:
cc @estroz